src/codegen.cpp

```diff
@@ -3726,10 +3726,22 @@ static jl_cgval_t emit_expr(jl_codectx_t &ctx, jl_value_t *expr)
                 mod = jl_globalref_mod(mn);
                 mn = (jl_value_t*)jl_globalref_name(mn);
             }
-            if (jl_symbol_name((jl_sym_t*)mn)[0] == '@')
-                jl_errorf("macro definition not allowed inside a local scope");
-            name = literal_pointer_val(ctx, mn);
-            bnd = jl_get_binding_for_method_def(mod, (jl_sym_t*)mn);
+            JL_TRY {
```

Comment:
```
julia> begin
         for i = 1:1 end
         macro g() end
       end
ERROR: error compiling anonymous: macro definition not allowed inside a local scope
```
vs.
```
julia> begin
         for i = 1:1 end
         macro g() end
       end
ERROR: macro definition not allowed inside a local scope
Stacktrace:
 [1] macro expansion at ./REPL[1]:3 [inlined]
 [2] anonymous at ./<missing>:?
```

Both results are wrong, but at least it now usually won't corrupt the C++ stack.